sql/parser/parse_test.go

```diff
@@ -275,9 +275,14 @@ func TestParse2(t *testing.T) {
 			`SELECT CAST('1' AS INT)`},
 		// Double negation. See #1800.
 		{`SELECT *,-/* comment */-5`,
-			`SELECT *, - -5`},
+			`SELECT *, - - 5`},
 		{"SELECT -\n-5",
-			`SELECT - -5`},
+			`SELECT - - 5`},
+		{"SELECT 1e-\n-1",
```

Comment:
This only tests the parser, if you tried to evaluate this it (hopefully) wouldn't work. Discarding this in the parser would require a lookahead. Since we've copied that from Postgres, better not to touch it.